### ISSUE TITLE
No error message when file_sd_configs include cannot be found

### ISSUE THREAD

**Issue by kaihendry** [prometheus/prometheus#1553]
Tested on 322ad49:

```
 wget http://s.natalian.org/2016-04-13/prometheus-missing.yml
 prometheus -config.file=prometheus-missing.yml
```

No complaint of [missing YML include](http://s.natalian.org/2016-04-13/1460525863_1912x1036.png).

Furthermore `promtool` doesn't detect anything is amiss either:

```
$ promtool check-config prometheus-missing.yml
Checking prometheus.yml
 SUCCESS: 0 rule files found
```

**Comment by grobie** [comment]
:+1: We should at least give a warning in promtool if the file is not present. While it's a valid config (the file might just be added later on by a different process), it would also help in debugging to warn about it during prometheus server start.

**Comment by Gouthamve** [comment]
So promtool now errors out instead of warning if the file is not in place:
```
➜  prometheus-1.7.1.linux-amd64 ./promtool check-config prometheus.yml
Checking prometheus.yml
  FAILED: "first.rules" does not point to an existing file
```

I think a warning is what was agreed upon. For anyone looking to tackle this:

promtool: Instead of returning an error here, logging it as warning would be better: https://github.com/prometheus/prometheus/blob/master/cmd/promtool/main.go#L96

prometheus: Logging a warning when the number of files matched is `0` would be good: https://github.com/prometheus/prometheus/blob/master/discovery/file/file.go#L82

**Comment by cstyan** [comment]
if you haven't started this already, I'll give it a shot @Gouthamve

**Comment by cstyan** [comment]
Fixed in #3062, will go out in 2.0. Should we mark this as fixed?

**Comment by Gouthamve** [comment]
Thanks @cstyan!
